assets/main.js

```diff
@@ -0,0 +1,90 @@
+import {displayStats} from "./displayStats.js";
+import {displayTodayStats} from "./displayStats.js";
+import {generateText} from "./generateText.js";
+import {markCurrentChar} from "./displayText.js";
+import {handleKeyDownEvent} from "./handleKeyDownEvent.js";
+import {incrementSeconds} from "./timeHandler.js";
+
+const generateTextButton = document.getElementById('generate-text-button');
+const resetProgressButton = document.getElementById('reset-button');
+const statsText = document.getElementById('last-set-stats-text');
+const statsTextForSeconds = document.getElementById('last-set-stats-time-text');
+const todayStatsText = document.getElementById('today-stats-text');
+let paragraphWithText = document.getElementById('text');
+
+let charArray = [];
+let charIndex = 0;
+let seconds = 0;
+let userKeyTypeCount = 0;
+let userMistakesCount = 0; 
+let amountOfSets = 0;
+let generateTextButtonIsClicked = false;
+statsTextForSeconds.innerHTML = 'Seconds: ' + seconds + 's';
+statsText.innerHTML = displayStats(0, 0);
+todayStatsText.innerHTML = displayTodayStats(0, 0, 0, 0);
+
+generateTextButton.addEventListener('click', async function() {
+  if (generateTextButtonIsClicked) {
+      resetProgress();
+      document.removeEventListener('keydown', keyDownHandler);
+  }
+
+  charArray = await generateText(paragraphWithText);
+
+  const incrementSecondsInterval = setInterval(function() {
+    seconds = incrementSeconds(seconds, statsTextForSeconds);
+    let minutes = seconds / 60;
+    const charactersPerMinute = userKeyTypeCount / minutes;
+
+    if (userKeyTypeCount == 0) {
+      statsText.innerHTML = 'CPM: ' + '0 ' + 'Wrong Chars: ' + '0%';
+    }
+    else {
+      statsText.innerHTML = 'CPM: '+ Math.round(charactersPerMinute) + ' ' + 'Wrong Chars: ' + Math.round((userMistakesCount * 100 / userKeyTypeCount * 100) / 100) + '%';
```

Comment:
```suggestion
      statsText.innerHTML = `CPM: ${Math.round(charactersPerMinute)} Wrong Chars: ${Math.round((userMistakesCount * 100 / userKeyTypeCount * 100) / 100)}%`;
```
[template-strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals?retiredLocale=de) might be easier to read